Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final:generate-code

### Describe the bug

My jenkins build failed with the error below after upgrading to `quarkus 3.2.0.Final`, the maven version is `3.9.1`

```
INFO] --- quarkus:3.2.0.Final:generate-code (default) @ lqa-service ---
[WARNING] Error injecting: io.quarkus.maven.GenerateCodeMojo
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) [Guice/ErrorInjectingConstructor]: NullPointerException: Cannot invoke "io.quarkus.devtools.messagewriter.MessageWriter.warn(String, Object[])" because "this.log" is null
  at BuildAnalyticsProvider.<init>(BuildAnalyticsProvider.java:35)
  at ClassRealm[plugin>io.quarkus.platform:quarkus-maven-plugin:Final, parent: URLClassLoader@723279cf]
      \_ installed by: WireModule -> PlexusBindingModule
  at BootstrapSessionListener.<init>(BootstrapSessionListener.java:26)
      \_ for 2nd parameter buildAnalyticsProvider
  at ClassRealm[plugin>io.quarkus.platform:quarkus-maven-plugin:Final, parent: URLClassLoader@723279cf]
      \_ installed by: WireModule -> PlexusBindingModule
  while locating BootstrapSessionListener
  while locating Object annotated with *
  while locating GenerateCodeMojo

Learn more:
  https://github.com/google/guice/wiki/ERROR_INJECTING_CONSTRUCTOR

1 error

======================
Full classname legend:
======================
....

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.655 s
[INFO] Finished at: 2023-07-13T02:21:32Z
[INFO] ------------------------------------------------------------------------
[INFO] [jenkins-event-spy] Generated ...abc.log
....

[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final:generate-code (default) on project xxx-service: Execution default of goal io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final:generate-code failed: Unable to load the mojo 'generate-code' (or one of its required components) from the plugin 'io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final': com.google.inject.ProvisionException: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) [Guice/ErrorInjectingConstructor]: NullPointerException: Cannot invoke "io.quarkus.devtools.messagewriter.MessageWriter.warn(String, Object[])" because "this.log" is null
[ERROR]   at BuildAnalyticsProvider.<init>(BuildAnalyticsProvider.java:35)
[ERROR]   at ClassRealm[plugin>io.quarkus.platform:quarkus-maven-plugin:Final, parent: URLClassLoader@723279cf]
[ERROR]       \_ installed by: WireModule -> PlexusBindingModule
[ERROR]   at BootstrapSessionListener.<init>(BootstrapSessionListener.java:26)
[ERROR]       \_ for 2nd parameter buildAnalyticsProvider
[ERROR]   at ClassRealm[plugin>io.quarkus.platform:quarkus-maven-plugin:Final, parent: URLClassLoader@723279cf]
[ERROR]       \_ installed by: WireModule -> PlexusBindingModule
[ERROR]   while locating BootstrapSessionListener
[ERROR]   while locating Object annotated with *
[ERROR]   while locating GenerateCodeMojo
[ERROR]   at ClassRealm[plugin>io.quarkus.platform:quarkus-maven-plugin:Final, parent: URLClassLoader@723279cf]
[ERROR]       \_ installed by: WireModule -> PlexusBindingModule
[ERROR]   while locating Mojo annotated with @Named("io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final:generate-code")
[ERROR] 
[ERROR] Learn more:
[ERROR]   https://github.com/google/guice/wiki/ERROR_INJECTING_CONSTRUCTOR
[ERROR] 
[ERROR] 1 error
[ERROR] 
[ERROR] ======================
[ERROR] Full classname legend:
[ERROR] ======================
[ERROR] BootstrapSessionListener: "io.quarkus.maven.components.BootstrapSessionListener"
[ERROR] BuildAnalyticsProvider:   "io.quarkus.maven.BuildAnalyticsProvider"
[ERROR] Final:                    "3.2.0.Final"
[ERROR] GenerateCodeMojo:         "io.quarkus.maven.GenerateCodeMojo"
[ERROR] Mojo:                     "org.apache.maven.plugin.Mojo"
[ERROR] Named:                    "com.google.inject.name.Named"
[ERROR] PlexusBindingModule:      "org.eclipse.sisu.plexus.PlexusBindingModule"
[ERROR] URLClassLoader:           "java.net.URLClassLoader"
[ERROR] WireModule:               "org.eclipse.sisu.wire.WireModule"
[ERROR] ========================
[ERROR] End of classname legend:
[ERROR] ========================
[ERROR] 
[ERROR]       role: org.apache.maven.plugin.Mojo
[ERROR]   roleHint: io.quarkus.platform:quarkus-maven-plugin:3.2.0.Final:generate-code
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
[Pipeline] }
```

### Output of `java -version`

17

### Quarkus version or git rev

3.2.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

maven 3.9.1, 3.9.2

Hi @aloubyansky , on Jenkins build only, more error is below.

```
Caused by: java.lang.NullPointerException: Cannot invoke "io.quarkus.devtools.messagewriter.MessageWriter.warn(String, Object[])" because "this.log" is null
    at io.quarkus.analytics.AnonymousUserId.write (AnonymousUserId.java:73)
    at io.quarkus.analytics.AnonymousUserId.loadOrCreate (AnonymousUserId.java:49)
    at io.quarkus.analytics.AnonymousUserId.<init> (AnonymousUserId.java:32)
    at io.quarkus.analytics.AnonymousUserId.getInstance (AnonymousUserId.java:21)
    at io.quarkus.analytics.AnalyticsService.<init> (AnalyticsService.java:82)
    at io.quarkus.maven.BuildAnalyticsProvider.<init> (BuildAnalyticsProvider.java:37)
```

--- a/independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java
+++ b/independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java
@@ -70,17 +70,22 @@ public class AnalyticsService implements AutoCloseable {
 
     final private RestClient restClient;
     final private ConfigService config;
-    final private AnonymousUserId userId;
+    final private AnonymousUserId anonymousUserId;
     final private MessageWriter log;
     final FileLocations fileLocations;
 
     public AnalyticsService(final FileLocations fileLocations, MessageWriter log) {
         this.fileLocations = fileLocations;
-        this.log = log;
+        if (log == null) {
+            this.log = MessageWriter.info();
+            this.log.info("No logger provided, using default");
+        } else {
+            this.log = log;
+        }
         this.postFutures = new ConcurrentLinkedQueue<>();
-        this.restClient = new RestClient(log);
-        this.userId = AnonymousUserId.getInstance(fileLocations, log);
-        this.config = new ConfigService(this.restClient, this.userId, fileLocations, log);
+        this.restClient = new RestClient(this.log);
+        this.anonymousUserId = AnonymousUserId.getInstance(fileLocations, this.log);
+        this.config = new ConfigService(this.restClient, this.anonymousUserId, fileLocations, this.log);
     }
 
     public void buildAnalyticsUserInput(Function<String, String> analyticsEnabledSupplier) {
@@ -100,7 +105,7 @@ public class AnalyticsService implements AutoCloseable {
             final Map<String, Object> context = createContextMap(applicationModel, buildInfo);
             sendIdentity(context);
             Track trackEvent = Track.builder()
-                    .userId(userId.getUuid())
+                    .userId(anonymousUserId.getUuid())
                     .context(context)
                     .event(trackEventType)
                     .properties(TrackProperties.builder()
@@ -150,9 +155,9 @@ public class AnalyticsService implements AutoCloseable {
     }
 
     void sendIdentity(final Map<String, Object> context) {
-        if (this.userId.isNew()) { // fixme when inactive on 1st call it will not send identity.
+        if (this.anonymousUserId.isNew()) { // fixme when inactive on 1st call it will not send identity.
             this.restClient.postIdentity(Identity.builder()
-                    .userId(this.userId.getUuid())
+                    .userId(this.anonymousUserId.getUuid())
                     .context(context)
                     .timestamp(Instant.now())
                     .build());

--- a/independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnonymousUserId.java
+++ b/independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnonymousUserId.java
@@ -29,8 +29,8 @@ public class AnonymousUserId {
 
     // Singleton
     private AnonymousUserId(final FileLocations fileLocations, final MessageWriter log) {
-        this.uuid = loadOrCreate(fileLocations.getUUIDFile());
         this.log = log;
+        this.uuid = loadOrCreate(fileLocations.getUUIDFile());
     }
 
     public String getUuid() {
@@ -42,13 +42,20 @@ public class AnonymousUserId {
     }
 
     private String loadOrCreate(Path file) {
-        if (Files.exists(file)) {
-            return load(file);
-        } else {
-            String uuid = UUID.randomUUID().toString();
-            write(uuid, file);
-            isNew = true;
-            return uuid;
+        try {
+            if (Files.exists(file)) {
+                return load(file);
+            } else {
+                String uuid = UUID.randomUUID().toString();
+                write(uuid, file);
+                isNew = true;
+                return uuid;
+            }
+        } catch (Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug("[Quarkus build analytics] Could not create UUID file at " + file.toAbsolutePath(), e);
+            }
+            return "N/A";
         }
     }
 
@@ -60,7 +67,9 @@ public class AnonymousUserId {
                     .map(String::trim)
                     .orElse("empty");
         } catch (IOException e) {
-            log.warn("[Quarkus build analytics] Could not read redhat anonymous UUID file at " + file.toAbsolutePath(), e);
+            if (log.isDebugEnabled()) {
+                log.debug("[Quarkus build analytics] Could not read redhat anonymous UUID file at " + file.toAbsolutePath(), e);
+            }
         }
         return uuid;
     }
@@ -70,8 +79,10 @@ public class AnonymousUserId {
             FileUtils.createFileAndParent(uuidFile);
             FileUtils.append(uuid, uuidFile);
         } catch (IOException e) {
-            log.warn("[Quarkus build analytics] Could not write redhat anonymous UUID to file at " + uuidFile.toAbsolutePath(),
-                    e);
+            if (log.isDebugEnabled()) {
+                log.debug("[Quarkus build analytics] Could not write redhat anonymous UUID to file at "
+                        + uuidFile.toAbsolutePath(), e);
+            }
         }
     }
 }

--- a/independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/rest/RestClient.java
+++ b/independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/rest/RestClient.java
@@ -113,9 +113,11 @@ public class RestClient implements ConfigClient, SegmentClient {
             }
             return Optional.empty();
         } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
-            log.warn("[Quarkus build analytics] Analytics remote config not received. " +
-                    e.getClass().getName() + ": " +
-                    (e.getMessage() == null ? "(no message)" : e.getMessage()));
+            if (log.isDebugEnabled()) {
+                log.debug("[Quarkus build analytics] Analytics remote config not received. " +
+                        e.getClass().getName() + ": " +
+                        (e.getMessage() == null ? "(no message)" : e.getMessage()));
+            }
         }
         return Optional.empty();
     }

--- a/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/AnalyticsServiceTest.java
+++ b/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/AnalyticsServiceTest.java
@@ -158,6 +158,16 @@ class AnalyticsServiceTest extends AnalyticsServiceTestBase {
         assertTrue(new File(FILE_LOCATIONS.getFolder().toString() + "/" + FILE_LOCATIONS.lastTrackFileName()).exists());
     }
 
+    @Test
+    void nullLogger() {
+        AnalyticsService service = new AnalyticsService(FILE_LOCATIONS, null);
+        assertNotNull(service);
+        service.sendAnalytics(TrackEventType.BUILD,
+                mockApplicationModel(),
+                Map.of(),
+                new File(FILE_LOCATIONS.getFolder().toUri()));
+    }
+
     private void assertMapEntriesNotEmpty(int size, Map<String, Object> map) {
         assertNotNull(map);
         assertEquals(size, map.size());

--- a/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/AnonymousUserIdManualTest.java
+++ b/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/AnonymousUserIdManualTest.java
@@ -16,7 +16,7 @@ import io.quarkus.analytics.config.FileLocationsImpl;
 import io.quarkus.devtools.messagewriter.MessageWriter;
 
 @Disabled("For manual testing purposes only")
-class UserIdManualTest {
+class AnonymousUserIdManualTest {
 
     private final FileLocations fileLocations = FileLocationsImpl.INSTANCE;
 

--- a/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/AnonymousUserIdTest.java
+++ b/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/AnonymousUserIdTest.java
@@ -1,0 +1,22 @@
+package io.quarkus.analytics;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.analytics.config.TestFileLocationsImpl;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+class AnonymousUserIdTest {
+
+    @Test
+    void createWithNullLogger() throws IOException {
+        TestFileLocationsImpl fileLocations = new TestFileLocationsImpl();
+        fileLocations.setUuidFile(null);
+        AnonymousUserId anonymousUserId = AnonymousUserId.getInstance(fileLocations, MessageWriter.info());
+        assertNotNull(anonymousUserId);
+    }
+
+}

--- a/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/config/TestFileLocationsImpl.java
+++ b/independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/config/TestFileLocationsImpl.java
@@ -8,7 +8,7 @@ import java.util.UUID;
 public class TestFileLocationsImpl implements FileLocations {
 
     private final Path tempDir;
-    private final Path uuidFile;
+    private Path uuidFile;
     private final Path remoteConfigFile;
     private final Path lastTryFile;
     private final Path localConfigFile;
@@ -57,6 +57,10 @@ public class TestFileLocationsImpl implements FileLocations {
     @Override
     public String lastTrackFileName() {
         return "lasttrack.json";
+    }
+
+    public void setUuidFile(Path uuidFile) {
+        this.uuidFile = uuidFile;
     }
 
     public void deleteAll() throws IOException {

